### ISSUE TITLE
Add a reporter that shows context and individual descriptions

### ISSUE THREAD

**Issue by stefanoborini** [r-lib/testthat#1006]
Currently, there is:

1. minimal reporter: shows only green dots
2. summary reporter: shows the context, followed by each context tests as green dots.
3. location reporter: does not show the context, shows the test the description and each individual assertions location

1 and 2 are very minimal and don't print the test_that description. 3 is too verbose. I tried various reporters and I could not find one that shows something like

```
context 1:
description of test 1: ....
description of test 2: ...F.
context 2:
description of test 1: .S...F...

```
Or

```
context 1/description of test 1: .....
context 1/description of test 2: .....
```

In other words, something intermediate between 2 and 3.

**Comment by krlmlr** [comment]
Does the default `reporter = "progress"` achieve what you're looking for?

**Comment by stefanoborini** [comment]
@krlmlr no. It just shows the context

**Comment by stefanoborini** [comment]
@hadley why close?

**Comment by hadley** [comment]
@stefanoborini it was 2 days ago and you hadn't replied so I assume you were no longer interested.

Given that you are still interested, this is unlikely something we would develop, but you could easily create your own reporter that does exactly what you want.

**Comment by stefanoborini** [comment]
@hadley Easter vacation. Please reopen, I will provide a PR later.

**Comment by hadley** [comment]
Sorry not to be clear — I meant this would be something you could maintain for yourself.

**Comment by jimhester** [comment]
For completeness, here is some code you could use, but I agree we don't need to add anything to testthat.

``` r
DetailedSummaryReporter <- R6::R6Class("DetailedSummaryReporter", inherit = testthat::SummaryReporter,
  public = list(
  start_test = function(context, test) {
    self$cat_tight(" ", test, ": ")
  },
  end_test = function(context, test) {
    self$cat_line()
  },
  start_context = function(context) {
    self$cat_tight(context, ":\n")
  },
  end_context = function(context) { }
  )
)

devtools::test("~/p/memoise", reporter = DetailedSummaryReporter$new())
#> Loading memoise
#> Testing memoise
#> filesystem:
#>  using a filesystem cache works: ...........
#>  two functions with the same arguments produce different caches (#38): ...
#> memoise:
#>  memoisation works: .............
#>  memoisation depends on argument: ...........
#>  interface of wrapper matches interface of memoised function: ...
#>  dot arguments are used for hash: ............
#>  default arguments are used for hash: .......
#>  default arguments are evaluated correctly: ........
#>  symbol collision: ........
#>  visibility: ..
#>  is.memoised: ...
#>  visibility: ..
#>  can memoise anonymous function: .....
#>  can memoise primitive: .....
#>  printing a memoised function prints the original definition: ..
#>  memoisation can depend on non-arguments: .............
#>  it fails if already memoised: .
#>  it evaluates arguments in proper environment: ..
#>  it does have namespace clashes with internal memoise symbols: ..
#>  arguments are evaluated before hashing: ...
#>  argument names don't clash with names in memoised function body: ..
#> has_cache:
#>  it works as expected with memoised functions: ...
#>  it errors with an un-memoised function: .
#> timeout:
#>  it stays the same if not enough time has passed: ..........
#> missing:
#>  it works with missing arguments: .....
#> s3:
#>  using a s3 cache works: S
#> 
#> ══ Skipped ═════════════════════════════════════════════════════════════════════
#> 1. using a s3 cache works (@test-s3.R#6) - Reason: No AWS Credentials
#> 
#> ══ DONE ════════════════════════════════════════════════════════════════════════
```

<sup>Created on 2020-04-14 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by stefanoborini** [comment]
@jimhester why not? this is how both python and junit show tests.

**Comment by stefanoborini** [comment]
@hadley I don't see why I must slap that code everywhere in my packages when it's basically a trivial addition that conforms with how the rest of the world shows tests in verbose mode.
